include/pagmo/problem.hpp

```diff
@@ -1761,8 +1762,17 @@ class problem
         return m_ptr.get();
     }
 
-    // A small helper to check if a vector containes unique elements.
-    template <typename U>
+    // A small helper to check if a vector containes unique elements. The version for arithmetic types
+    // is also protected against possible nans
+    template <typename U, enable_if_is_floating_point<U> = 0>
+    static bool all_unique(std::vector<U> x)
+    {
+        std::sort(x.begin(), x.end(), [](const U &el1, const U &el2) { return detail::less_than_f(el1, el2); });
+        auto it = std::unique(x.begin(), x.end());
```

Comment:
So maybe the question is, do we need that nan == nan is true? Philosophy ... so for the time being I would leave as is